sympy/matrices/common.py

```diff
@@ -2065,6 +2066,8 @@ def __pow__(self, num):
                     except (AttributeError, MatrixError):
                         pass
                 return a._eval_pow_by_recursion(num)
+            elif num.is_Number!=True and num.is_negative==None and a.det()==0:
+                return Pow.__new__(type(a),a,num)
```

Comment:
type(a) will be Matrix, but this should create a MatPow. I think you can just create MatPow here directly.

Reply:
Also, put spaces after commas.